routing/router_test.go

```diff
@@ -1949,12 +1961,19 @@ func TestDisconnectedBlocks(t *testing.T) {
 	// Create a 15 block fork. We first let the chainView notify the router
 	// about stale blocks, before sending the now connected blocks. We do
 	// this because we expect this order from the chainview.
+	ctx.chainView.notifyStaleBlockAck = make(chan struct{}, 1)
 	for i := len(minorityChain) - 1; i >= 0; i-- {
 		block := minorityChain[i]
 		height := uint32(forkHeight) + uint32(i) + 1
 		ctx.chainView.notifyStaleBlock(block.BlockHash(), height,
-			block.Transactions)
+			block.Transactions, t)
+		<-ctx.chainView.notifyStaleBlockAck
 	}
+	ctx.chainView.notifyStaleBlockAck = nil
+
+	time.Sleep(time.Second * 2)
```

Comment:
Do we still need this now that we're acking? We should be all caught up at this stage.

Reply:
So when I remove this particular sleep, sometimes the test fails about 50% of the time with:

--- FAIL: TestDisconnectedBlocks (1.63s)
    router_test.go:2006: found edge in graph

I'm guessing this is because it takes a little time for the graph to be updated when reaching here? https://github.com/lightningnetwork/lnd/blob/c93824ec9a4ade4e011c8010bc7c484fcb71e5c9/routing/router.go#L1123